[Game] added a FishFinder job:

NL0bP opened the pull request:
- with the help of crutches added the work of the echo sounder; [Scripts] Added echo sounder test command:
- /fishfinder set {true|false]

[Game] Updated echo sounder operation:
- renamed FishFinderManager to FishSchoolManager;
- now the echo sounder is turned on/off by R;
- echo sounder turns off when you press T;

NL0bP committed:
[Game] added a FishFinder job:
- with the help of crutches added the work of the echo sounder;
[Scripts] Added echo sounder test command:
- /fishfinder set {true|false]

[Game] Updated echo sounder operation:
- renamed FishFinderManager to FishSchoolManager;
- now the echo sounder is turned on/off by R;
- echo sounder turns off when you press T;

ZeromusXYZ commented on AAEmu.Game/Core/Managers/World/SpawnManager.cs:
```diff
@@ -517,6 +517,12 @@ public void SpawnAll()
                     }
                     _log.Info("{0} Doodads spawned", count);
+
+                    if (worldId == 0)
```
Can't we do this in a better way?
Especially, since instances can also have their own fishing pools. (e.g. Mirage fishing)

ZeromusXYZ commented on AAEmu.Game/Models/Game/World/Interactions/Use.cs:
```diff
@@ -19,6 +24,19 @@ public class Use : IWorldInteraction
             }
+
+            // TODO ID=21902, View Fish Finder: Scan around with the Fish Finder. Detected schools of fish will be displayed on the map.
+            if (skillId == 21902)
```
Make this a const somewhere instead of a magic number

ZeromusXYZ reviewed (changes requested):
We should make sure that the shown spots are actually in the same instance as the player.
To be fair, we need to fix this overall for all this too :)
The loading at WorldId == 0 is probably not a good idea, as it's possible to not have a world id 0.
Just start it like any other manager during load?

NL0bP committed:
[Game] FishSchoolManager changes:
- added constants for Doodads, Buffs, Skills;
- load the list of fishing spots for each world separately;

ZeromusXYZ commented on AAEmu.Game/Models/Game/Skills/SkillConstants.cs:
```diff
@@ -0,0 +1,7 @@
+{
+    public enum SkillConstants : uint
+    {
+        ViewFishFinder = 21902,
```
We already have a file for skill constants, just add it there
``AAEmu.Game/Models/StaticValues/SkillsEnum.cs``
It doesn't really need to be a enum.

NL0bP committed:
[Game] Moved the ViewFishFinder constant to SkillEnum;
[Game] forgot to turn off Buff SearchSchoolOfFish if we start moving;
